hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml

```diff
@@ -3982,6 +3982,17 @@
   </description>
 </property>
 
+<property>
+  <name>dfs.datanode.delete.corrupt.replica.from.disk.enable</name>
+  <value>true</value>
```

Comment:
If the default value is true, there is a risk of block missing according to [HDFS-16985](https://issues.apache.org/jira/browse/HDFS-16985). I suggest setting the default value to false, as block missing is a more serious problem than disk file deletion delay. What's your opion?

Reply:
Thanks @zhangshuyan0 for your comment.

From the DataNode point of view, if  already confirmed the meta file or data file is lost. it should be deleted directly from the memory and disk and this is expected behavior.

For HDFS-16985 mentioned, if the current cluster deployment adopts the AWS EC2 + EBS solution, can adjust `dfs.datanode.delete.corrupt.replica.from.disk.enable` is false as needed to avoid deleting disk data.

So I think it might be better from datanode perspective default to set `dfs.datanode.delete.corrupt.replica.from.disk.enable` to true

looking forward to your suggestions again.

Reply:
Hi @zhangshuyan0 Would you mind look it again, thanks~